app/controllers/carts_controller.rb

```diff
@@ -7,4 +8,24 @@ def show
     @show_comments = true
   end
 
+  def index
+    role = 'requester'
+    my_carts = Cart.joins(:approvals).where(:approvals => {:role => role, :user_id => @current_user[:id]})
+    if my_carts.empty?
+      role = 'approver'
+      my_carts = Cart.joins(:approvals).where(:approvals => {:role => role, :user_id => @current_user[:id]})
+    end
+    @closed_carts = my_carts.closed
+    @open_carts = my_carts.open
+    render :template => "carts/index", :locals => {:open_carts => @open_carts,
+                                                   :closed_carts => @closed_carts,
+                                                   :user => @current_user, :role => role,
+                                                   limit: CLOSED_CART_LIMIT}
```

Comment:
The idiomatic Rails thing would be to leave the `render` implicit and passing all of the models via instance variables. If you prefer setting the local variables explicitly, I would make them regular variables within the controller action.